builtin/clean.c

```diff
@@ -167,6 +171,29 @@ static int remove_dirs(struct strbuf *path, const char *prefix, int force_flag,
 		goto out;
 	}
 
+	if (is_mount_point(path)) {
+#ifndef CAN_UNLINK_MOUNT_POINTS
+		if (!quiet) {
+			quote_path_relative(path->buf, prefix, &quoted);
+			printf(dry_run ?
+			       _(msg_would_skip_mount_point) :
+			       _(msg_skip_mount_point), quoted.buf);
+		}
+		*dir_gone = 0;
+#else
+		if (!dry_run && unlink(path->buf)) {
```

Comment:
If this is a dry run, it will give a warning about a failed remove and return?

Reply:
No, if it is _not_ a dry-run, and if the `unlink()` failed, _then_ it will report a warning and return.

Reply:
Ah, sorry for the confusion.